.github/workflows/tests.yml

```diff
@@ -0,0 +1,73 @@
+name: tests
+
+on: push
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  rust:
+    name: rust build
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build
+        working-directory: p2panda-rs
+        run: cargo build --verbose
+
+      - name: Run tests
+        working-directory: p2panda-rs
+        run: cargo test --verbose
+  js:
+    name: js build
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Read node version from .nvmrc
+        working-directory: p2panda-js
+        id: nvmrc
+        run: echo ::set-output name=NODE_VERSION::$(cat .nvmrc)
+
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.nvmrc.outputs.NODE_VERSION }}"
+
+      - name: Cache node modules and cargo registry
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - run: npm ci
```

Comment:
Nitpicking: Can be good to give all job steps names in the same manner as in line 49!

Reply:
Agreed, a bit of explanation never hurts. I often omit the names for commands that are also used outside Github actions like `npm ci` or `npm start`.